src/main/java/graphql/schema/idl/SchemaTypeDirectivesChecker.java

```diff
@@ -160,8 +196,219 @@ private boolean isNoNullArgWithoutDefaultValue(InputValueDefinition definitionAr
         return definitionArgument.getType() instanceof NonNullType && definitionArgument.getDefaultValue() == null;
     }
 
-    private boolean argValueMatchesAllowedType(TypeDefinitionRegistry typeRegistry, Value instanceValue, Type allowedArgType) {
-        // this code is not written yet and is tricky.
-        return true;
```

Comment:
Would you agree its tricky? ;)

Reply:
Agreed! We could even contribute back to graphql-js for this one.